app/controllers/images_controller.rb

```diff
@@ -8,21 +8,26 @@ def index
 
   def create
     if Image.create(image_params)
-      redirect_to camp_images_path(params.permit(:camp_id))
+      redirect_to event_camp_images_path(@event, @camp)
     else
       render action: :index
     end
   end
 
   def destroy
     @camp.images.find(params[:id]).destroy!
-    redirect_to camp_images_path(params.permit(:camp_id))
+    redirect_to event_camp_images_path(@event, @camp)
   end
 
   private
 
   def load_camp!
-    @camp = Camp.find(params[:camp_id])
+    @event = Event.find(params[:event_id])
+    not_found if @event.nil? if @event.nil?
```

Comment:
You might consider pulling out load_camp! and load_event! into a module so you don't have to define them in multiple places; this loading code is important, reusable, and shouldn't be defined in more than one place.

At the very least you should drop the double `if @event.nil?` here :)